pkg/controllers/job/job_controller.go

```diff
@@ -139,6 +131,10 @@ func (cc *jobcontroller) Initialize(opt *framework.ControllerOption) error {
 	cc.errTasks = newRateLimitingQueue()
 	cc.recorder = recorder
 	cc.workers = workers
+	cc.maxRequeueNum = opt.MaxRequeueNum
+	if cc.maxRequeueNum < 0 {
```

Comment:
please check this parameter to make sure it is larger than zero :)

Reply:
Not sure if I fully understand what you mean. Do you mean `cc.maxRequeueNum` can't be negative? l think the controller should retry forever when `MaxRequeueNum` is set to negative